src/utilstrencodings.cpp

```diff
@@ -229,7 +229,7 @@ vector<unsigned char> DecodeBase64(const char* p, bool* pfInvalid)
 string DecodeBase64(const string& str)
 {
     vector<unsigned char> vchRet = DecodeBase64(str.c_str());
-    return (vchRet.size() == 0) ? string() : string((const char*)&vchRet[0], vchRet.size());
+    return std::string((const char*)vchRet.data(), vchRet.size());
```

Comment:
The question is whether calling `std::string(garbage_pointer, 0)` is well-defined. I can't tell from what cppreference.com says (*The behavior is undefined if s does not point at an array of at least count elements of CharT, including the case when s is a null pointer.*).

As an alternative, what about `std::string(vchRet.begin(), vchRet.end())`?